File: 프로그래머스/2/17684. ［3차］ 압축/［3차］ 압축.js
function solution(msg) {
    let result = [];
    let alphaDict = {};
    let alphaDictLen = 0;
    
    for(let i = 1 ; i <= 26 ; i++){
        alphaDict[String.fromCharCode(64 + i)] = i;
        alphaDictLen++;
    }
    
    let s = 0, e = 0;
    while(true){
        e += 1
        if(e === msg.length){
            result.push(alphaDict[msg.slice(s, e)]);
            break;
        }
        
        if(!(msg.slice(s, e+1) in alphaDict)){
            alphaDict[msg.slice(s, e+1)] = ++alphaDictLen;
            result.push(alphaDict[msg.slice(s, e)]);   
            s = e;
        }
    }
    return result;
}

File: 프로그래머스/2/17684. ［3차］ 압축/［3차］ 압축.py
def solution(msg):
    result = []
    alpha_dict = {chr(64 + i): i for i in range(1, 27)}
    
    s = e = 0
    while True:
        e += 1
        if e == len(msg):
            result.append((alpha_dict[msg[s : e]]))
            break
        
        if msg[s : e+1] not in alpha_dict:
            alpha_dict[msg[s : e+1]] = len(alpha_dict) + 1
            result.append(alpha_dict[msg[s:e]])
            s = e
    return result
            